File: 0066-plus-one/0066-plus-one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        changed = True
        i = len(digits)-1
        while changed:
            changed = False
            digits[i]+=1
            if digits[i]%10 == 0:
                changed = True
                digits[i] = 0
                i-=1
                if i<0: 
                    digits.insert(0,0) 
                    i+=1
        return digits
        

File: 0066-plus-one/0066-plus-one.java
class Solution {
    public int[] plusOne(int[] digits) {
        // String s = "";
        // for(int d: digits){
        //     s = s+d;
        // }
        // String newNum = String.valueOf(Integer.valueOf(s)+1);
        // int[] ans = new int[newNum.length()];
        // for (int i = 0; i < newNum.length(); i++) {
        //     ans[i] = newNum.charAt(i) - '0';
        // }
        // return ans;

        String s = "";
        for (int d : digits) {
            s = s + d;
        }

        StringBuilder newNum = new StringBuilder(s);
        int i = newNum.length() - 1;
        while (i >= 0 && newNum.charAt(i) == '9') {
            newNum.setCharAt(i, '0');
            i--;
        }

        if (i >= 0) {
            newNum.setCharAt(i, (char)(newNum.charAt(i) + 1));
        } else {
            newNum.insert(0, '1');
        }

        int[] ans = new int[newNum.length()];
        for (i = 0; i < newNum.length(); i++) {
            ans[i] = newNum.charAt(i) - '0';
        }

        return ans;
    }
}